src/Capgemini.DataMigration.Core/Cache/SimpleMemoryCache.cs

```diff
@@ -41,5 +41,20 @@ protected TItem GetCachedItem(string cacheKey)
                 return newIstance;
             }
         }
+
+        protected TItem TryGetCachedItem(string cacheKey)
+        {
+            var cachedObject = MemoryCache.Default.Get(cacheKey, null);
+            return (TItem)cachedObject;
+        }
```

Comment:
We should not change the abstract class as not all classes which inherit from it require this new method and we do not want to change its main purpose and one activity it is doing - this abstract base class is meant to be used by scenarios where if object does not exist in the cache is retrieved and added to cache for the next call to be there. As a new method is added then some classes which use it must be changed as well to throw an exception that it is not supposed to be used. I recommend to create a new dedicated class for this purpose, which may inherit from this class and adds the additional layer , then your implementation can inherit from that, or refactor the code and use the existing class

Reply:
I don't totally agree, but get your point.  I've now created a new DictionaryCache class for use in the code that I've written where the calling or containing code takes more responsibility for populating and managing the cache.  (Technically we could refactor the mapping cache to self-populate, but this would be a risk as it would mean a significant restructure of some fairly delicate looking code, and the entity cache has to be pre-populated)  I've also added a new BaseCache class shared between the two base implementations, although at the moment there's nothing in this - we may want to add shared memory management logic in the future.  In the case of the entity data, the use of a dictionary rather than a System.Runtime.Caching class has significantly reduced the cache memory footprint because this cache contains a large number of small records.

Reply:
(For info Mike TryGetValue is the name of the corresponding method System.Collections.Generic.IDictionary)

Reply:
> (For info Mike TryGetValue is the name of the corresponding method System.Collections.Generic.IDictionary)

Yes but that method is a test that returns false if it doesn't find anything and outputs a value. This just returns null and isn't really different from a standard get method. Granted a try-catch isn't the only approach to this (and probably not the best way), you could happily also implement something like the TryGetValue dictionary functionality, it depends how you want to use this method really. I just don't see where the try element of this is.

Reply:
I'm not sure if it's the convention, but the `TryX` methods that I've seen have a `bool` return type and an `out` parameter. You can then combine it with an `if` statement e.g.

```csharp
if (int.TryParse("10", out int result))
{
   // Work with result
}
```